app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -87,8 +92,31 @@ def import_all(pulp_ids = nil, options = {})
         end
       end
 
+      def pulp3_import_all(pulp_ids = nil, options = {})
+        fail _("Importing and indexing all content units is not supported in Pulp3") if pulp_ids.blank?
+        index_repository_association = options.fetch(:index_repository_association, true) && self.manage_repository_association
+        service_class = SmartProxy.pulp_master!.content_service(content_type, pulp3_support?)
+        process_block = lambda do |unit|
+            unit = unit.with_indifferent_access
+            model = Katello::Util::Support.active_record_retry do
+              self.where(:pulp_id => unit['_href']).first_or_create
+            end
+            service = service_class.new(model.pulp_id)
+            service.backend_data = unit
+            service.update_model(model)
+          if index_repository_association
+            unit.slice('_href', 'repository_memberships')
```

Comment:
with pulp3 we can't index repository association as part of this, i think we should remove that feature even with pulp2.

Reply:
Removed for bopth pulp2 and pulp3..